big-number/src/main/java/BigNumber.java

```diff
@@ -0,0 +1,230 @@
+        if (absValue.length() > num.absValue.length()) {
+            return (this.positive) ? 1 : -1;
+        }
+        // both has the same sigh, but this has fewer digits
+        if (absValue.length() < num.absValue.length()) {
+            return (this.positive) ? -1 : 1;
+        }
+
+        // both has the same sigh and same number of digits
+        if (absValue.length() == num.absValue.length()) {
+            for (int i = 0; i < absValue.length(); i++) {
+                int digitThis = charToDigit(this.absValue.charAt(i));
+                int digitNum = charToDigit(num.absValue.charAt(i));
+                if (digitThis > digitNum) {
+                    return (this.positive) ? 1 : -1;
+                }
+                if (digitThis < digitNum) {
+                    return (this.positive) ? -1 : 1;
+                }
+            }
+        }
+        // this and num are equal
+        return 0;
+    }
+
+    public String toString() {
+        return (positive) ? absValue : "-" + absValue;
+    }
+
+    // addition of two positive numbers
+    private String digitsPlus(String num1, String num2) {
+
+        StringBuilder resultSB = new StringBuilder();
+
+        if (num1.length() > num2.length()) {
+            num2 = addSomeZeros(num2, num1.length());
+        }
+        if (num1.length() < num2.length()) {
+            num1 = addSomeZeros(num1, num2.length());
+        }
+
+        int addTens = 0;
+        for (int i = num1.length() - 1; i >= 0; i--) {
+            int next = charToDigit(num1.charAt(i)) + charToDigit(num2.charAt(i)) + addTens;
+            if (next < 10) {
+                resultSB.append(next);
+                addTens = 0;
+            } else {
+                resultSB.append(next % 10);
+                addTens = 1;
+            }
+        }
+        resultSB.trimToSize();
+        return resultSB.reverse().toString();
+    }
+
+    // subtraction of two positive numbers
+    private String digitsMinus(String num1, String num2) {
+
+        StringBuilder resultSB = new StringBuilder();
+
+        if (num1.length() > num2.length()) {
+            num2 = addSomeZeros(num2, num1.length());
+        }
+        if (num1.length() < num2.length()) {
+            num1 = addSomeZeros(num1, num2.length());
+        }
+
+        int addUnits = 0;
+        int subTens = 0;
+        for (int i = num1.length() - 1; i >= 0; i--) {
+            int next = charToDigit(num1.charAt(i)) - subTens - charToDigit(num2.charAt(i));
+            if (next < 0) {
+                subTens = 1;
+                addUnits = 10;
+            } else {
+                subTens = 0;
+            }
+            next += addUnits;
+            resultSB.append(next);
+            addUnits = 0;
+        }
+
+        int i = resultSB.length() - 1;
+        while (resultSB.charAt(i) == '0' && resultSB.length() > 1) {
+            resultSB.deleteCharAt(i);
+            i--;
+        }
+        resultSB.trimToSize();
```

Comment:
my idea was the following: when we subtract two very long numbers, for example, 500 characters, but our answer is very short, we should set free the memory allocated for the StringBuilder
in our case there is no need to use it, because StringBuilder is a local variable, we will return a short String in our method and memory will be cleared automatically